miner/worker.go

```diff
@@ -799,6 +805,19 @@ func (w *worker) commitTransaction(env *environment, tx *types.Transaction) ([]*
 	if tx.Type() == types.BlobTxType {
 		return w.commitBlobTransaction(env, tx)
 	}
+
+	// If a conditional is set, check prior to applying
+	if conditional := tx.Conditional(); conditional != nil {
+		txConditionalMinedTimer.UpdateSince(conditional.SubmissionTime)
+
+		if err := env.header.CheckTransactionConditional(conditional); err != nil {
+			return nil, fmt.Errorf("failed tx conditional: %w", errTxConditionalInvalid)
+		}
+		if err := env.state.CheckTransactionConditional(conditional); err != nil {
+			return nil, fmt.Errorf("failed tx conditional: %w", errTxConditionalInvalid)
+		}
+	}
+
```

Comment:
Can we add a block-building flag to rate-limit conditional-txs in the final synchronous block building work? Or some safety mechanism, in case the load is too much to sequence?

And ideally, a rate-limit expressed in condition-cost, not just number of txs. E.g. use the standard rate limit Go util, create a burst of X (set to maximum tx conditional cost), and throughput rate of Y (the configured rate), and then take from the bucket with the `cond.Cost()`. And don't block on rate limit, but just drop the tx if there's no capacity available, in favor of non-conditional txs. I think this would be a helpful anti-DoS measure.

Reply:
Yeah good idea! Added a flag for the rate limit here. Set it to be 5tps*max cost so 5000